types/propEq.d.ts

```diff
@@ -1,6 +1,29 @@
-export function propEq<T>(val: T): {
-  <K extends PropertyKey>(name: K): (obj: Record<K, T>) => boolean;
-  <K extends PropertyKey>(name: K, obj: Record<K, T>): boolean;
+import { Placeholder } from 'ramda';
+
+export function propEq(__: Placeholder): never;
+export function propEq<V>(val: V): {
+  <K extends PropertyKey>(name: K): (obj: Record<K, any>) => boolean;
+  <U extends Record<PropertyKey, any>>(__: Placeholder, obj: U): (name: keyof U) => boolean;
+  <K extends keyof U, U extends Record<PropertyKey, any>>(name: K, obj: U): boolean;
 };
-export function propEq<T, K extends PropertyKey>(val: T, name: K): (obj: Record<K, T>) => boolean;
-export function propEq<K extends keyof U, U>(val: U[K], name: K, obj: U): boolean;
+export function propEq<K extends PropertyKey>(__: Placeholder, name: K): {
+  (__: Placeholder, obj: Record<K, any>): <V>(val: V) => boolean;
+  <V>(val: V, obj: Record<K, any>): boolean
+  <V>(val: V): (obj: Record<K, any>) => boolean
+};
+export function propEq<V, K extends PropertyKey>(val: V, name: K): (obj: Record<K, any>) => boolean;
+
+
+export function propEq<U extends Record<any, any>>(__: Placeholder, ___: Placeholder, obj: U): {
+  (__: Placeholder, name: keyof U): (__: Placeholder) => boolean;
```

Comment:
Should the final return not be `never`? That would match the intention of the very first overload

Reply:
Thank you for noticing. You are right, it should be `never` here